nlp_apps.ipynb

```diff
@@ -797,11 +772,251 @@
+    "    \"company\",\n",
+    "    \"fruit\",\n",
+    "    \"fruit\",\n",
+    "    \"fruit\",\n",
+    "]\n",
+    "\n",
+    "class_0 = \"company\"\n",
+    "class_1 = \"fruit\"\n",
+    "\n",
+    "test_data = [\n",
+    "    \"Apple Inc. supplier Foxconn demos its own iPhone-compatible smartwatch\",\n",
+    "    \"I now know how to make a delicious apple pie thanks to the best teachers ever\"\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Step 2:- Preprocessing the dataset**\n",
+    "\n",
+    "In this step, we would be doing some preprocessing on the dataset like breaking the sentence into words and converting to lower case.\n",
+    "\n",
+    "We already have a `words(sent)` function defined in `text.py` which does the task of splitting the sentence into words."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data_processed = [words(i) for i in train_data]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Step 3:- Feature Extraction from the text**\n",
+    "\n",
+    "Now we would be extracting features from the text like extracting the set of words used in both the categories i.e. `company` and `fruit`.\n",
+    "\n",
+    "This frequency of a word would help in calculating the probability of that word being in a particular class. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of words in `company` class: 49\n",
+      "Number of words in `fruit` class: 49\n"
+     ]
+    }
+   ],
+   "source": [
+    "words_0 = []\n",
+    "words_1 = []\n",
+    "\n",
+    "for sent, tag in zip(train_data_processed, train_target):\n",
+    "    if(tag == class_0):\n",
+    "        words_0 += sent\n",
+    "    elif(tag == class_1):\n",
+    "        words_1 += sent\n",
+    "    \n",
+    "print(\"Number of words in `\" + class_0 + \"` class:\", len(words_0))\n",
+    "print(\"Number of words in `\" + class_1 + \"` class:\", len(words_1))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As you might have observed that our dataset is equally balanced i.e. we have an equal number of words in both the classes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Step 4:-Making the Naive Bayes Model**\n",
+    "\n",
+    "Using Naive Bayes classifier we can calculate the probability of a word in `company` and `fruit` class and then multiplying all of them to get the probability of that sentence belonging each of the given classes. But if a word is not in our dictionary then this leads to the probability of that class becoming zero. For eg:- the word *Foxconn* is not in the dictionary of any of the classes. Due to this the \n",
```

Comment:
Add a 'the' before 'Naive Bayes'. Also, add a 'to' before 'each of the given classes'. Also, you don't need to write 'eg:-', just write 'example'.

Finally, the paragraph seems to end abruptly.